src/java/org/apache/cassandra/schema/Schema.java

```diff
@@ -446,6 +446,12 @@ public TableMetadata getTableMetadata(String keyspace, String table)
     @Override
     public TableMetadata getTableMetadata(TableId id)
     {
+        if (distributedKeyspaces == null || localKeyspaces == null)
+        {
+            logger.info("Likely due to mocked tests, null values for distributedKeyspaces ({}), and localKeyspaces ({})",
+                        distributedKeyspaces, localKeyspaces);
+            return null;
+        }
```

Comment:
I wonder if this is necessary and if it is better to instead improve the test? I think mixing in logic intended for tests into production code should be avoided. When it's necessary, I think I've seen examples that are protected by some flag to indicate a test is in progress, e.g. `if (isTesting && (distributedKeyspaces == null || localKeyspaces == null))`.

If a cndb test mocks `Schema`, then perhaps something like `when(mockSchema.getTableMetadata(any()).thenReturn(null)` could be used to prevent an NPE? That is, assuming a `null` value for this method is acceptable for the test which seems to be the case here.

Another alternative is for getters to replace the direct access to `localKeyspaces` and `distributedKeyspaces` below; then the getters that can be mocked as well.

Reply:
@djatnieks It is not related to mock. The NPE happened here on `Schema.instance` object, which is not mocked.